### ISSUE TITLE
NodeModel.js gives null value for NodeId at handy-owl server

### ISSUE THREAD

**Issue by rushirajnenuji** [NCEAS/metacatui#570]
@amoeba and I have been working on using the already retrieved the information about the member node from endpoint: `https://cn.dataone.org/cn/v2/node/` 

We're trying to inject the current node's NodeId into the header as JSON-LD along with other information. The NodeModel.js has an array variable `members` which contains all the member node's information retrieved from the above endpoint. The source code for the JSON-LD element looks like this:
```
for (var i = 0; i < this.nodeModel.attributes.members.length; i++) {
    if(this.nodeModel.attributes.members[i].identifier == this.nodeModel.attributes.currentMemberNode) {
        var nodeModelObject = this.nodeModel.attributes.members[i];
    }
}

let elJSON = {
    "@context": {
        "@vocab": "http://schema.org/"
    },
    "@type": "DataCatalog",
    "description": nodeModelObject.description,
    "identifier": nodeModelObject.identifier,
    "image": nodeModelObject.logo,
    "name": nodeModelObject.name,
    "url": nodeModelObject.url
};

```

However, the currentMemberNode attribute is `null` for the case of MetaCatUI deployed at `handy-owl`

**Observation**:
In the NodeModel.js at/around `Line 160` we have:
`var thisMember = _.findWhere(thisModel.get("members"), { baseURL:  (MetacatUI.appModel.get("baseUrl") + MetacatUI.appModel.get('context') + MetacatUI.appModel.get("d1Service")).replace("/v2", "").replace("/v1", "") });`
The node identifier of the current Member Node MetacatUI is pointing to can always be found at `{base_url}/v2/node` but this line of code is looking for the node in the Coordinating Node’s nodelist.

**Potential Solution**:
This section of NodeModel could trust the local MetaCat's nodeId instead of querying the CN.

@csjx , @laurenwalker : We were wondering if this section of NodeModel.js requires a change in the code, or if there would be a better approach to achieve the results.

**Comment by mbjones** [comment]
That would be a nice change for nodes that are not registered in a CN. It would require making sure that the dataone properties are filled out without registering, which currently can't be done through the admin application -- so metacat.properties would need to be modified directly.  Maybe there should be an admin button to save the properties without registering in DataONE.

**Comment by laurenwalker** [comment]
In the AppModel in MetacatUI, there is a `nodeId` property which can be set to the local metacat's nodeId, so that feature already exists. There is also a property called `title` which is meant to be a repository title/node title.

However, the other node properties like description, logo, etc. do not exist in MetacatUI yet. I think it would be a good idea to add a default MN Node object for those properties. We could do that when we refactor the MetacatUI configuration. See #281.

As a side note, the code snippet you pasted here that has the `for` loop is accessing the Backbone model properties via model.attributes. Model properties should be accessed via the Model.get() function - http://backbonejs.org/#Model-get

**Comment by amoeba** [comment]
Thanks for chiming in @mbjones and @laurenwalker!

@laurenwalker, when you say:

> In the AppModel in MetacatUI, there is a nodeId property which can be set to the local metacat's nodeId, so that feature already exists. 

Do you mean that you expect `nodeId` to be set to the local Metacat's nodeId or that, theoretically, it could be? AFAICT the current behavior is that `MetacatUI.appModel.get("nodeId")` is `null` for the default theme because it's defined as `null` in the base `AppModel.js`. I see that it's `urn:node:CN` for the dataone theme. And it's `null` for the arctic theme. That all kinda makes sense.

Overall, I'm curious if there's ever a case where the `nodeId` wouldn't be what's at `MNCore.getCapabilities()` (https://handy-owl.nceas.ucsb.edu/metacat/d1/mn/v2/node) and, if not, if we can always just use that instead of querying the CN for it?

**Comment by laurenwalker** [comment]
We could get the node info from d1/mn/v2/node as a backup if the node is not found on the CN node list at cn/v2/node. We need to grab the CN node list anyway (to do things like build the MN filter list, show MN icons in search results, etc.) so it makes sense to try to find the current node info there, if we can. If not, we could grab the local node info and use that.

**Comment by amoeba** [comment]
I can dig that. @rushirajnenuji what do you think about adding that functionality?

**Comment by rushirajnenuji** [comment]
Yes, @amoeba @laurenwalker - that should solve the problem we're having. Thanks!

**Comment by csjx** [comment]
Also address serializing correct node values described in https://github.com/NCEAS/metacatui/issues/797#issuecomment-428766584.
